src/Domain/User.php

```diff
@@ -1,39 +1,41 @@
-<?php
+<?php declare(strict_types=1);
 
 namespace Becklyn\Security\Domain;
 
 use Becklyn\Ddd\Events\Domain\EventProvider;
-use Tightenco\Collect\Support\Collection;
+use Symfony\Component\Security\Core\User\PasswordAuthenticatedUserInterface;
+use Illuminate\Support\Collection;
 
 /**
  * @author Marko Vujnovic <mv@201created.de>
+ *
  * @since  2020-03-02
  */
-interface User extends EventProvider
+interface User extends EventProvider, PasswordAuthenticatedUserInterface
```

Comment:
Our Domain classes may not implement or extend Symfony classes. The domain layer may not reference Symfony or any other infrastructural code. This we we have a separate SymfonyUser class in the infrastructure, which may do so.

If this PasswordAuthenticatedUserInterface is needed, please move it to the SymfonyUser interface instead.